File: csharp/GildedRose/GildedRoseApprovalTests.cs
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace csharp {
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class GildedRoseApprovalTests {

        private List<Item> createItemList(string itemName, int sellIn, int quality) {
            return new List<Item> { new Item { Name = itemName, SellIn = sellIn, Quality = quality } };
        }  
        
        [Test]
        public void OneItemApproval() {
            // Arrange
            GildedRose sut = new GildedRose(createItemList("Sulfuras, Hand of Ragnaros", 20, 80));
    
            // Act
            sut.UpdateQuality();
    
            // Assert
            // The Verify call creates a file named GildedRoseApprovalTests.OneItemApproval.received.txt
            // and compares it against the Approved result file, GildedRoseApprovalTests.OneItemApproval.approved.txt
            // If the contents of the two files match, the test passes
            // otherwise the test fails and the diff of the two files is shown
            Approvals.Verify(sut.Items[0].ToString());
        }
    }
}

File: java/gildedrose_covered/src/test/java/gildedrose/GildedRoseApprovalTests.java
package gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class GildedRoseApprovalTests {

    @Test
    public void test_UpdateQuality_SingleItem_AtSellin_NoQuality() {
        Item[] items = new Item[]{new Item("normal", 0, 0)};
        GildedRose sut = new GildedRose(items);
        sut.updateQuality();
        assertEquals("normal, -1, 0", sut.items[0].toString());
        Approvals.verify(sut.items[0].toString());
    }

    @Test
    public void test_UpdateQuality_SingleItem() {
        CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                new String[] {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer[] {-1, 0, 5, 10, 11},    // SellIn
                new Integer[] { 0, 1, 49, 50 }       // Quality
        );
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose sut = new GildedRose(items);
        sut.updateQuality();
        return sut.items[0].toString();
    }
}
